feature/repair/src/main/java/com/iteneum/repair/presentation/RepairView.kt

```diff
@@ -63,17 +88,19 @@ fun RepairView() {
                 enabled = false,
                 label = stringResource(id = R.string.label_unit),
                 hint = stringResource(id = R.string.hint_unit),
-                onValueChanged = {/* TODO - Unit Text field - To verify if extra functionality required */ }
+                onValueChanged = {
+                    repairViewModel.getUnitDepartmentFromView(it)
+                }
             )
             LPPhoneNumberText(
                 modifier = Modifier.padding(
                     top = sizes.extraSize14
                 ),
-                value = contactPhoneNumberValue,
+                value = repairViewModel.repairModel.contactPhone,
```

Comment:
You are not observing the contatPhone when it changes

Reply:
I did not get this point, sorry. I thougth that is showing correctly. Because i have it on two parameters.

value (So it can print the phoneNumber when inserting/deleting a number
![image](https://user-images.githubusercontent.com/25538206/236572244-4b94bffe-c0e0-48e9-b64b-637169517f2a.png)

onPhoneChanged (So it can be updated when the user inserts another value)
![image](https://user-images.githubusercontent.com/25538206/236572321-e37393a9-aac8-4f1a-aacf-8c8a38219a59.png)

Reply:
Well, I mean, that you need to create an remember in this way: 
viewModel: 
 `private val _uiState = mutableStateOf(UiStateMainActivity(data = test, loading = true))
    val uiState: State<UiStateMainActivity> = _uiState`
    
    ` fun updateLoading(){
        _uiState.value = _uiState.value.copy(
            loading = !_uiState.value.loading
        )
    }`
Compose: 
`private val viewModel by viewModels<MainActivityViewModel>()`
`val uiState by remember { viewModel.uiState }`
`Button(
                        onClick = { viewModel.updateLoading() },
                        modifier = Modifier.padding(bottom = 20.dp)
                    ) {
                        Text(text = "Click ${uiState.loading}")
                    }`

Reply:
Ok, will make the change

Reply:
Modified. Done! (Help provided by Carlos)